rudder-license/src/main/scala/com/normation/license/License.scala

```diff
@@ -0,0 +1,724 @@
+   * In rawContent, we don't want to change number of lines, order,
+   * or anything else from which depends the signature, and thus any
+   * cleaning/filtering operation must be done before that point
+   * (if they make sense).
+   */
+  def readLicenseInformation(contentLines: List[String]): Maybe[(LicenseInformation, String)] = {
+    //utility to parse a version with the expected format
+    def parseVersion(version: String) = Version.from(version) match {
+      case Some(v) => Right(v)
+      case None    => Left(LicenseError.Parsing(s"Error: '${version}' can't be parsed as a software version"))
+    }
+
+    val rawContent = contentLines.mkString("\n")
+    val content = parseKeyValues(contentLines)
+
+    // license mandatory fields
+    val missingFields = LicenseField.InformationKeys.allValues.filterNot(k => content.keySet.contains(k.name))
+
+    if(missingFields.nonEmpty) {
+      Left(LicenseError.Parsing(s"The license file is missing mandatory fields: ${missingFields.map(_.name).mkString(", ")}"))
+    } else {
+      //we know that all the fields are their, so we can use themap("key") directly
+      import LicenseField._
+
+      for {
+        startDate  <- ISODate.read(content(InformationKeys.StartDate.name))
+        endDate    <- ISODate.read(content(InformationKeys.EndDate.name))
+        minVersion <- parseVersion(content(InformationKeys.MinVersion.name))
+        maxVersion <- parseVersion(content(InformationKeys.MaxVersion.name))
+      } yield {
+        val info = LicenseInformation(
+            Licensee(content(InformationKeys.Licensee.name))
+          , SoftwareId(content(InformationKeys.SoftwareId.name))
+          , MinVersion(minVersion)
+          , MaxVersion(maxVersion)
+          , StartDate(startDate)
+          , EndDate(endDate)
+        )
+        // return the couple
+        (info, rawContent)
+      }
+    }
+  }
+
+
+
+
+  /*
+   * Read a signed license from a file and check format (but not signature)
+   */
+  def readLicense(licensePath: String): Maybe[RudderLicense.Unchecked] = {
+    for {
+      fileContent <- readFile(licensePath)
+      license     <- parseLicense(fileContent)
+    } yield {
+      license
+    }
+  }
+
+}
+
+/**
+ * Read public/private RSA key
+ */
+final object RSAKeyManagement {
+  import MaybeLicenseError._
+
+  //bouncy castle provider
+  Security.addProvider(new BouncyCastleProvider())
+
+  /**
+   * the two methods are highly similar but differ just sufficiently
+   * (SubjectPublicKeyInfo/PrivateKeyInfo, keyFactory.generatePublic/generatePrivate...)
+   * to be more work to factorize than duplicate :(
+   */
+
+  def readPKCS8PublicKey(path: String): Maybe[RSAPublicKey] = {
+    for {
+                    // the parser able to read PEM files
+                    // Parser may be null if the key is invalid
+      parser     <- {
+                      val p = new PEMParser(Files.newBufferedReader(Paths.get(path)))
```

Comment:
No need to check if the path exists ?

Reply:
And provide a better error message if does not exists!

Reply:
you're right

Reply:
Ah sorry, in fact it is actually checked:
```
{
....
} withError { ex => LicenseError.Crypto(s"Error when trying to read the private key at '${path}'. The key may be invalid (expecting PKCS8 format): ${ex.getMessage}")
}
```